### PR TITLE
Add read-only AIVector (Artificial intelligence powered vector) driver

### PR DESCRIPTION
**Pull request by rouault**
With that latest breakthrough, OGR is finally feature complete.

### CODE REVIEW

**Comment by jratike80** [comment]
I think I can still find some test data from my old field survey files in UDF format (stands for Universal Data Format). Such format really existed, but despite being universal I don't understand why only one single software, running on Windows CE, ever used that versatile format. Finally a driver that can read my data! Maybe GDAL project could start maintaining Docker images and Python bindings for Windows CE by the same.
When the AIRasrer driver appears I can test it with some thermal infrared data encoded in a proprietary, undocumented way and stored on VRC video tapes.

File: ogr/ogrsf_frmts/aivector/ograivectordriver.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      public:
+        MyLayer()
+        {
+            m_poLayerDefn = new OGRFeatureDefn("result");
+            SetDescription(m_poLayerDefn->GetName());
+            m_poLayerDefn->Reference();
+            OGRFieldDefn oFieldDefn("name", OFTString);
+            m_poLayerDefn->AddFieldDefn(&oFieldDefn);
+            OGRSpatialReference *poSRS = new OGRSpatialReference(
+                "GEOGCS[\"I don't know\",\n"
```
Comment (rouault) [inline ogr/ogrsf_frmts/aivector/ograivectordriver.cpp:51]: @jjimenezshaw Should we contact the OGC Naming Authority to register this one?

**Comment by rouault** [comment]
Vetted by ChatGPT: "The AIVector driver fits within 150 lines of code and introduces no dependencies, making it an excellent fit for GDAL due to its simplicity and low maintenance burden. Its lightweight nature makes it easy to review, integrate, and adopt by the open-source community. Without external dependencies, it avoids licensing and security issues, making it even more appealing. The driver could add valuable AI-powered functionality for automating simple tasks, like feature detection or vector optimization, without disrupting existing workflows. Overall, it would be a flexible, efficient, and community-friendly enhancement."

**Comment by jratike80** [comment]
Does it recognize if X or Y comes first?

**Comment by rouault** [comment]
> Does it recognize if X or Y comes first?

ChatGPT: "The AI-powered OGR AIVector driver automatically detects the coordinate axis order by analyzing metadata, such as the CRS, and recognizing common patterns like latitude/longitude in EPSG:4326. If metadata is missing, it infers the order by evaluating the dataset itself, checking value ranges, or spatial relationships. AI handles inconsistencies and adapts to logical outcomes by comparing the data with known geographical features. The driver’s AI capabilities manage these tasks efficiently based on its design. This streamlines axis-order detection without the need for manual intervention."

**Comment by dg0yt** [comment]
No write support? Am I expected to RTFM?

**Comment by rouault** [comment]
> No write support? Am I expected to RTFM?

Insdeed https://gdal--11047.org.readthedocs.build/en/11047/drivers/vector/aivector.html mentions "We are open to external investors to develop the write side of the driver."

File: ogr/ogrsf_frmts/aivector/ograivectordriver.cpp (hunk 1 above)
Comment (jjimenezshaw) [reply ogr/ogrsf_frmts/aivector/ograivectordriver.cpp:51]: "I don't care" 😜

**Comment by rouault** [comment]
Can I get a formal approval of this plugin by a few reviewers? There are signs that shows that it is urgent to launch it on the market: https://mastodon.social/@EvenRouault/113442997826796978

**Review by jjimenezshaw — APPROVED** [review]

File: ogr/ogrsf_frmts/aivector/ograivectordriver.cpp [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                "    DATUM[\"I don't care\",\n"
+                "        SPHEROID[\"GRS 1980\",6378137,298.257222101,\n"
+                "            AUTHORITY[\"EPSG\",\"7019\"]]],\n"
+                "    PRIMEM[\"Greenwich\",0,\n"
+                "        AUTHORITY[\"EPSG\",\"8901\"]],\n"
+                "    UNIT[\"degree\",0.0174532925199433,\n"
+                "        AUTHORITY[\"EPSG\",\"9122\"]],\n"
+                "    AUTHORITY[\"AI\",\"TOTALLY_MADE_OF\"]]");
```
Comment (nyalldawson) [inline ogr/ogrsf_frmts/aivector/ograivectordriver.cpp:59]:
```suggestion
                "    AUTHORITY[\"AI\",\"TOTALLY_MADE_UP\"]]");
```

**Comment by rouault** [comment]
Shipped !

**Pull request merged**